pkg/reconciler/managed/reconciler.go

```diff
@@ -525,6 +525,41 @@ func (r *Reconciler) Reconcile(_ context.Context, req reconcile.Request) (reconc
 		"external-name", meta.GetExternalName(managed),
 	)
 
+	// If managed resource has a deletion timestamp and and a deletion policy of
+	// Orphan, we do not need to observe the external resource before attempting
+	// to unpublish connection details and remove finalizer.
+	if meta.WasDeleted(managed) && managed.GetDeletionPolicy() == xpv1.DeletionOrphan {
+		log = log.WithValues("deletion-timestamp", managed.GetDeletionTimestamp())
+		managed.SetConditions(xpv1.Deleting())
+
+		if err := r.managed.UnpublishConnection(ctx, managed, ConnectionDetails{}); err != nil {
```

Comment:
Passing empty `ConnectionDetails{}` here is a little strange, but the only "unpublish" implementation we have right now is a no-op and allows secrets to be handled with GC. However, it is unclear whether we would ever really need to pass connection details to an unpublish connection, as the contents of the secret are not really important to unpublishing the entire secret. I presume the reason why the connection publisher interface accepts this argument is to allow an implementation that partially unpublishes connection details or verifies that details match before unpublishing. We could consider just removing the `ConnectionDetails` argument from `UnpublishConnection`, but that would technically be a breaking API change (though I doubt any providers would be significantly impacted if at all).

Reply:
The intent of the `UnpublishConnection` interface is supposed to be "remove this subset of connection details", without any opinion as to where those details are being unpublished from (e.g. a secret). Roughly the inverse of `PublishConnection`.

> Passing empty ConnectionDetails{} here is a little strange, but the only "unpublish" implementation we have right now is a no-op and allows secrets to be handled with GC.

This seems like a great thing to include in a comment on this call.

Reply:
@negz done :+1: